HttpURI toURI passes all info

gregw opened the pull request:
Fix #11465 and #7750
HttpURI.toURI user and fragment are retained.
Use to URI(String) constructor, as all URI constructors will parse the URI anyway.

gregw committed:
HttpURI toURI passes all info

Fix #11465 and #7750
HttpURI.toURI user and fragment are retained.
Use to URI(String) constructor, as all URI constructors will parse the URI anyway.

cowwoc reviewed (commented):
Consider using URI.create(String)

cowwoc commented on jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java:
```diff
@@ -265,8 +264,7 @@ default URI toURI()
         {
-            String query = getQuery();
-            return new URI(getScheme(), null, getHost(), getPort(), getPath(), query == null ? null : UrlEncoded.decodeString(query), null);
+            return new URI(toString());
```
Consider using `URI.create(String)` instead if you're okay with throwing `IllegalArgumentException` instead of `RuntimeException`.

gregw committed:
HttpURI toURI passes all info

Fix #11465 and #7750
HttpURI.toURI user and fragment are retained.
Use to URI(String) constructor, as all URI constructors will parse the URI anyway.

joakime commented:
EEEK!  This was bad!

joakime commented:
The multi-parameter `new URI(.....)` constructors will encode the various parts before they are added to the URI.
We don't want that.
